### DISCUSSION TITLE
Generate Audio with Python Script

### DISCUSSION THREAD

**Discussion by delmenhorst** [acids-ircam/RAVE#208] [General]
Hey,
I trained several models with Rave1, does anyone have any good resources on how to generate audio files just with a Python script? My goal is to generate some WAV files with the prior model.

Has anyone done this? I know that there was some script around, but I didn't get through. Would anyone be willing to share their experience?

cheers. :)

**Comment by OliverThomas2000** [comment]
```python
import torch
import librosa as li
import soundfile as sf
​
torch.set_grad_enabled(False)
​
model = torch.jit.load("your_model.ts").eval()
​
x = li.load("some_audio.wav")[0]
​
x = torch.from_numpy(x).reshape(1,1,-1)
z = model.encode(x)
​
z[:, 0] += torch.linspace(-2,2,z.shape[-1])
​
y = model.decode(z).numpy().reshape(-1)
​
sf.write("model_output.wav", y, 44100)
```

Hope this helps!

**Comment by delmenhorst** [comment]
thank you very much. 
Do you have also some code snippets for the prior model? :)

**Comment by m3da9ti** [comment]
I was wondering if there is a code already for unconditional generation vs latent alteration which the above code is for...
